src/Operation/InsertOne.php

```diff
@@ -94,6 +105,15 @@ public function __construct(string $databaseName, string $collectionName, $docum
             unset($options['writeConcern']);
         }
 
+        if (isset($options['codec'])) {
+            $document = $options['codec']->encodeIfSupported($document);
+
+            // Psalm's assert-if-true annotation does not work with unions, so
+            // assert the type manually instead of using is_document
+            // See https://github.com/vimeo/psalm/issues/6831
+            assert(is_array($document) || is_object($document));
```

Comment:
Interesting that you remembered to nest the assertion within the `if` condition here.

Reply:
At least I'm consistently inconsistent at some point. Or is it called "eventually consistent"? 🤔